contrib/pai_vscode/src/pai/yamlJobConfigCompletionProvider.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License in the project root for license information.
+ * @author Microsoft
+ */
+
+import * as fs from 'fs';
+// tslint:disable-next-line: no-require-imports
+import fuzzysearch = require('fuzzysearch');
+import * as yaml from 'js-yaml';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+interface IYamlJobConfigSnippet {
+    readonly name: string;
+    readonly label: string;
+    readonly documentation: string;
+    readonly insertText: string;
+}
+
+/**
+ * An OpenPAI completion provider provides yaml code snippets for job config.
+ */
+export class YamlJobConfigCompletionProvider implements vscode.CompletionItemProvider {
+    private snippets: IYamlJobConfigSnippet[] = [];
+
+    public constructor() {
+        this.loadSnippets();
+    }
+
+    public provideCompletionItems(document: vscode.TextDocument, position: vscode.Position): vscode.CompletionItem[] {
+        const word: string = document.getText(document.getWordRangeAtPosition(position));
```

Comment:
I think vscode's completion provider already uses fuzzy search to filter the result.
Maybe you can just return all snippets and let vscode do the filtering.